spyder/api/plugins/new_api.py

```diff
@@ -781,302 +1211,444 @@ def get_command_line_options(self):
+    :meth:`find_previous` and :meth:`replace` methods will be called
+    to handle the corresponding actions.
+
+    If set to ``False``, the default, the corresponding actions fall back
+    to calling these methods on the :guilabel:`Editor`.
+
+    Individual actions can be disabled with the
+    :meth:`spyder.plugins.application.plugin.enable_search_action` method
+    in the :guilabel:`Application` plugin.
+    """
 
     # ---- API: Available signals
     # -------------------------------------------------------------------------
     sig_focus_changed = Signal()
     """
-    This signal is emitted to inform the focus of this plugin has changed.
+    Report the focus state of this plugin has changed.
     """
 
     sig_toggle_view_changed = Signal(bool)
     """
-    This action is emitted to inform the visibility of a dockable plugin
-    has changed.
+    Report that visibility of a dockable plugin has changed.
 
     This is triggered by checking/unchecking the entry for a pane in the
-    `Window > Panes` menu.
+    :menuselection:`Window --> Panes` menu.
 
     Parameters
     ----------
     visible: bool
-        New visibility of the dockwidget.
+        Whether the dockwidget has been shown (``True``) or hidden (``False``).
     """
 
     sig_switch_to_plugin_requested = Signal(object, bool)
     """
-    This signal can be emitted to inform the main window that this plugin
-    requested to be displayed.
+    Request the main window show this plugin's widget.
 
-    Notes
-    -----
-    This is automatically connected to main container/widget at plugin's
+    This is automatically connected to main container/widget at plugin
     registration.
+
+    Parameters
+    ----------
+    plugin: SpyderDockablePlugin
+        The plugin object to show.
+    force_focus: bool | None
+        If ``True``, always give the plugin's widget focus when showing
+        or hiding it with this method. If ``None``, only give it focus when
+        showing, not hiding (i.e. setting ``state`` to ``True``),
+        and only if :attr:`RAISE_AND_FOCUS` is ``True``. If ``False``,
+        the default, don't give it focus regardless.
     """
 
     sig_update_ancestor_requested = Signal()
     """
-    This signal is emitted to inform the main window that a child widget
-    needs its ancestor to be updated.
+    Notify the main window that a child widget needs its ancestor updated.
     """
 
     # ---- Private methods
     # -------------------------------------------------------------------------
-    def __init__(self, parent, configuration):
+    def __init__(
+        self,
+        parent: spyder.app.mainwindow.MainWindow,
+        configuration: spyder.config.manager.ConfigurationManager,
+    ):
+        """
+        Initialize the plugin object (called automatically by Spyder).
+
+        .. important::
+
+            Plugins are initialized automatically by Spyder; plugins shouldn't
+            call or override this method directly.
```

Comment:
```suggestion
            Plugins are initialized automatically by Spyder, so they shouldn't
            call or override this method directly.
```

Same suggestion as for SpyderPluginV2.